packages/@sanity/default-layout/src/navbar/search/hooks/useSearch.ts

```diff
@@ -17,32 +18,46 @@ import {
   switchMap,
   tap,
 } from 'rxjs/operators'
-import {FINDABILITY_MVI} from '../constants'
-import {hasSearchableTerms} from '../contexts/search/selectors'
-import {SearchHit, SearchState} from '../types'
+import {searchClient} from '../../../versionedClient'
+import {SearchState} from '../types'
+import {hasSearchableTerms} from '../utils/hasSearchableTerms'
+import {getSearchableOmnisearchTypes} from '../utils/selectors'
+
+interface SearchRequest {
+  options?: SearchOptions
+  terms: SearchTerms
+}
 
 const INITIAL_SEARCH_STATE: SearchState = {
+  error: null,
   hits: [],
   loading: false,
-  error: null,
+  options: {},
   terms: {
     query: '',
     types: [],
   },
-  searchString: '',
 }
 
 function nonNullable<T>(v: T): v is NonNullable<T> {
   return v !== null
 }
 
-function sanitizeTerms(terms: SearchTerms) {
+function sanitizeRequest(request: SearchRequest) {
   return {
-    ...terms,
-    query: terms.query.trim(),
+    ...request,
+    terms: {
+      ...request.terms,
+      query: request.terms.query.trim(),
+    },
   }
 }
 
+const searchWeighted = createWeightedSearch(getSearchableOmnisearchTypes(schema), searchClient, {
```

Comment:
This line wont port to v3 since its using the global schema. In v3 this must be done inside useSearch and the useSchema hook. Possibly think about how it can be memoized there now.